core/src/main/java/hudson/tasks/CommandInterpreter.java

```diff
@@ -99,6 +118,7 @@ public boolean perform(AbstractBuild<?,?> build, Launcher launcher, TaskListener
                 Util.displayIOException(e, listener);
                 e.printStackTrace(listener.fatalError(Messages.CommandInterpreter_CommandFailed()));
             }
+            this.build = null;
```

Comment:
And <code>transient</code>... I doubt such approach will work with <code>SimpleBuilder</code>s